api/api-environments-v2/src/main/java/com/thoughtworks/go/apiv2/environments/representers/EnvironmentVariableRepresenter.java

```diff
@@ -31,12 +32,16 @@ public static void toJSON(OutputWriter outputWriter, EnvironmentVariableConfig e
     }
 
     public static EnvironmentVariableConfig fromJSON(JsonObject jsonReader) {
-        String name = jsonReader.get("name").getAsString();
-        String value = jsonReader.get("value").getAsString();
         boolean secure = jsonReader.has("secure") && jsonReader.get("secure").getAsBoolean();
-
-        EnvironmentVariableConfig environmentConfig = new EnvironmentVariableConfig(new GoCipher(), name, value, secure);
-        return environmentConfig;
+        EnvironmentVariableConfig environmentVariableConfig = new EnvironmentVariableConfig();
+        environmentVariableConfig.setName(jsonReader.get("name").getAsString());
+        if (secure) {
+            environmentVariableConfig.setEncryptedValue(jsonReader.get("encrypted_value").getAsString());
+            environmentVariableConfig.setIsSecure(secure);
+        } else {
+            environmentVariableConfig.setValue(new VariableValueConfig(jsonReader.get("value").getAsString()));
```

Comment:
Can't someone provide:
```
{
"name": "e",
"secure": "true",
"value": "to-be-encrypted-by-go"
}
```
and this should return
```
{
"name": "e",
"secure": "true",
"encrypted_value": "AES:encrypted"
}
```
because its a pain to use the encryption API to get the encrypted value and then use it in the environments API? 

Also, isn't https://github.com/gocd/gocd/pull/5967 trying to do the same?

Reply:
@varshavaradarajan didn't see that there was already a PR for this. I'll close this PR